helix-view/src/document.rs

```diff
@@ -582,6 +589,10 @@ impl Document {
         // and error out when document is saved
         self.path = path;
 
+        if self.text.len_bytes() > BIG_FILE_THRESHOLD {
+            self.enable_syntax_parser = false;
+        }
```

Comment:
I think we should let users have some feedback, otherwise they might be surprised on why there is suddenly no syntax.

Reply:
Makes sense to me, although when I was implementing this PR, I didn't find any easy way to implement such feature. I will try to look into this again.

Reply:
You might have to create a prompt UI element, which could also be helpful for other things.